services/onboarding-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/OnboardingService.java

```diff
@@ -835,8 +835,8 @@ public APIGatewayProxyResponseEvent provisionTenant(Map<String, Object> event, C
             try {
                 CreateStackResponse cfnResponse = cfn.createStack(CreateStackRequest.builder()
                         .stackName(stackName)
-                        .onFailure("DO_NOTHING")
-                        .timeoutInMinutes(60)
+                        .onFailure("DO_NOTHING") // This was set to DO_NOTHING to ease debugging of failed stacks. Maybe not appropriate for "production". If we change this we'll have to add a whole bunch of IAM delete permissions to the execution role.
+                        //.timeoutInMinutes(60) // Some resources can take a really long time to light up. Do we want to specify this?
```

Comment:
So I see two ways to address this, a long term and short term solution.

In the long term, we should be able to have a decent idea of what resources are about to be lit up by the CloudFormation stack we're about to create. Experimentally (either through our own data collection or through anonymous user-approved data collection) we can build an expectation of how long resources *typically* take to light up, and set our timeouts to be something considerably higher than that such that a CreateStack won't hang effectively indefinitely.

In the short term though, I agree that removing the timeout makes the most sense. Is it likely or possible that we create a CFT that never finishes? What's the likelihood that a stack taking longer than expected indicates a problem we would be able to act on (and not a service issue)? I'd argue both of these are significantly less likely than us creating a stack that takes longer than a random timeout and leads to a bad customer experience.